tutorial/src/index.css

```diff
@@ -1,111 +1,124 @@
-/* reset */
-* {
-  margin: 0;
-  padding: 0;
-  border: 0;
-  line-height: 1.5;
-}
+@import url("@shared/styles.css");
 
 body {
-  font-family: system-ui, "Segoe UI", Roboto, "Helvetica Neue", helvetica,
-    sans-serif;
+  align-items: center;
+  background-color: var(--color-n12);
+  color: var(--color-white);
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  min-height: 100vh;
 }
 
 main {
-  padding-top: 1em;
-  padding-bottom: 1em;
-  width: min(800px, 95vw);
-  margin: 0 auto;
+  width: 100vw;
+  height: 100vh;
+  display: flex;
+  flex-direction: column;
+}
+@media (min-width: 50rem) {
+  main {
+    border: 1px solid var(--color-p5);
+    width: 40rem;
+    height: min(60rem, 95vh);
+  }
 }
 
-h1 {
-  text-align: center;
-  margin-bottom: 8px;
-  font-size: 1.8em;
-  font-weight: 500;
+header {
+  border-bottom: 1px solid var(--color-p5);
+  padding: var(--size-md);
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
 }
 
-.badge {
-  text-align: center;
-  margin-bottom: 16px;
-}
-.badge span {
-  background-color: #212529;
-  color: #ffffff;
-  border-radius: 6px;
-  font-weight: bold;
-  padding: 4px 8px 4px 8px;
-  font-size: 0.75em;
+header span {
+  background: var(--color-p4);
+  padding: var(--size-xs);
+  border-radius: var(--size-md);
+  font-size: var(--size-sm);
 }
 
-ul {
-  margin: 8px;
-  border-radius: 8px;
-  border: solid 1px lightgray;
-  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
+h1 {
+  font-size: var(--size-2xl);
+  font-family: var(--font-family-display);
+  font-weight: 500;
+  font-stretch: var(--font-stretch-max);
 }
 
-ul:empty {
-  display: none;
+ul {
+  flex-grow: 1;
+  padding: var(--size-md);
+  list-style: none;
+  overflow-y: auto;
```

Comment:
Previously, the app would grow in height until the entire page required scrolling. It now has a fixed height (never more than the viewport) so that the message input is always in view. But this does mean that once there are many messages the newest messages appear at the bottom, scrolled out of view.

It's easy enough to add a `useEffect` to scroll to the bottom of the messages list. I just wasn't sure if we wanted to include that in the tutorial or keep this very focused on working with Convex rather than the details of building a UI like this.

Reply:
Eh, I'd skip. Lets keep the demo code as simple as possible. I think thats actually more important than how good a chat app it is.